### PR TITLE
feat(compaction): define rpc for interaction between rpc proxy and compactors

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: proto/hummock.proto [hunk 1]
```diff
@@ -435,6 +435,49 @@ message SubscribeCompactionEventResponse {
...
+  // DispatchCompactionTaskRequest is used to pass compaction task level parameters.
+  oneof task {
+    CompactTask compact_task = 1;
+    VacuumTask vacuum_task = 2;
+    FullScanTask full_scan_task = 3;
+    ValidationTask validation_task = 4;
+    CancelCompactTask cancel_compact_task = 5;
+  }
+
+  uint32 block_size_kb = 6;
```
Comment (hzxa21) [inline proto/hummock.proto:448]: block_size is provdied via opts right now so there is no need to put it here.

File: proto/hummock.proto [hunk 2]
```diff
@@ -435,6 +435,49 @@ message SubscribeCompactionEventResponse {
...
+  uint64 compactor_max_task_multiplier = 7;
```
Comment (hzxa21) [inline proto/hummock.proto:449]: this is providded via config right now so there is no need to put it here

File: proto/hummock.proto [hunk 3]
```diff
@@ -435,6 +435,49 @@ message SubscribeCompactionEventResponse {
...
+  // Set a watermark SST id to prevent full GC from accidentally deleting SSTs for in-progress write op.
+  uint64 start_sst_id = 8;
```
Comment (hzxa21) [inline proto/hummock.proto:451]: This should be maintained by proxy so there is no need to put it here,

**Comment by wcy-fdu** [comment]
merge to https://github.com/risingwavelabs/risingwave/pull/11904

**Pull request closed without merging**